akka-projection-grpc/src/main/protobuf/akka/projection/grpc/event_producer.proto

```diff
@@ -206,13 +206,20 @@ message Offset {
   // If empty it is assumed to be the persistence_id -> seq_nr of enclosing Event
   // or FilteredEvent.
   repeated PersistenceIdSeqNr seen = 2;
+  // Offsets by slice. If this is being used, then other offset fields will be empty.
+  repeated OffsetBySlice offsetsBySlice = 3;
```

Comment:
Would be cleaner if Offset contained a oneof with TimestampOffset and TimestampOffsetBySlice, and able to be extended to other offset types. But adding in a protocol compatible way instead.

Reply:
This is fine. One compatible alternative might be to change fields of Offset to repeated Offset field, and only have a `int32 slice` addition here? Maybe that doesn't work because undefined int32 becomes 0, which is the same as if slice=0.

Reply:
Since `optional` has been reintroduced, could have an `optional int32 slice` which will be `Option[Int]` for scalapb. I'll add the changes, and we can see if that's preferable. Still a bit more complicated on detecting it.

Reply:
Updated in https://github.com/akka/akka-projection/pull/1181/commits/3f7b45da9053a4c27fa729f36ace97cd6826f223. With offset changing from Option to Seq, touches a few more areas.